### ISSUE TITLE
Time units defined in dataset.yaml don't make it into the output NetCDF file

### ISSUE THREAD

**Issue by mkieburtzpnnl** [tsdat/tsdat#137]
**Describe the bug**
If a time coordinate is defined in dataset.yaml such as
```yaml
coords:
  time:
    attrs:
      long_name: Time (UTC)
      units: Seconds since 1970-01-01 00:00:00
```
The output file will discard the `units` value. Replacing it with `Seconds since 1970-01-01`

**To Reproduce**
Minimal set of steps to reproduce the behavior:
```
1. Open an output NetCDF file for a pipeline that has time units set in dataset.yaml other than `Seconds since 1970-01-01`
2. Note that the time units have changed to `Seconds since 1970-01-01`
```

**Expected behavior**
Time units in the output NetCDF file should be as defined in dataset.yaml

**Environment** 

- OS: Windows 10
- `tsdat`: 
```
Name: tsdat
Version: 0.5.9
Summary: A data processing framework used to convert time series data into standardized format.
Home-page: https://github.com/tsdat/tsdat
Author: tsdat
Author-email: tsdat@pnnl.gov
License: Simplified BSD (2-clause)
Location: c:\users\kieb843\anaconda3\envs\tsdat-pipelines\lib\site-packages
Requires: act-atmos, boto3, boto3-stubs, cftime, dunamai, h5netcdf, jsonpointer, netCDF4, numpy, pandas, pyarrow, pydantic, pyyaml, rich, tstring, typer, types-PyYAML, xarray, zarr
Required-by:
```

**Comment by maxwelllevin** [comment]
Hey @mkieburtzpnnl, thanks for bringing this to our attention. I ran through an ingest pipeline in the debugger to try and isolate where the extra units string is being dropped (the `00:00:00` bit), and it seems like it is preserved all the way up to the `dataset.to_netcdf(filepath, **to_netcdf_kwargs)` call in the netcdf writer class:

<img width="1271" alt="netcdf_kwargs_are_correct" src="https://user-images.githubusercontent.com/24307537/235201505-5a530a6e-8535-498c-b7a9-ce732f00740d.png">

However, it appears the `HH:MM:SS` string gets chopped off from the data that are actually written to disk. Shown below is a screenshot of that using default `xarray` settings to read the file, but it also seems to be the case using other utilities to read netcdf files.

<img width="436" alt="reopened_dataset_is_incorrect" src="https://user-images.githubusercontent.com/24307537/235201602-d9c1852b-92b2-4033-a45f-f41be53c68f0.png">

Interestingly, if you modify the `HH:MM:SS` string to be anything but `00:00:00`, it seems to work just fine:

<img width="1029" alt="image" src="https://user-images.githubusercontent.com/24307537/235202276-0da8b775-b9ea-4412-ba98-c026bca517bd.png">


In short, this is very likely an upstream issue in `xarray` that we don't have the ability to fix. While this does affect the units string for time, fortunately it does not actually affect the values or validity of the data being read/written and only applies when the HHMMSS part is all 0.
